skate-plugin/project-gen/src/jvmMain/kotlin/slack/tooling/projectgen/circuitgen/CircuitGenUi.kt

```diff
@@ -0,0 +1,181 @@
+import androidx.compose.foundation.lazy.items
+import androidx.compose.foundation.lazy.rememberLazyListState
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.Stable
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.awt.ComposePanel
+import androidx.compose.ui.unit.dp
+import com.slack.circuit.foundation.Circuit
+import com.slack.circuit.foundation.CircuitContent
+import com.slack.circuit.runtime.ui.ui
+import java.nio.file.Path
+import javax.swing.JComponent
+import org.jetbrains.jewel.foundation.theme.JewelTheme
+import org.jetbrains.jewel.ui.Orientation
+import org.jetbrains.jewel.ui.component.Checkbox
+import org.jetbrains.jewel.ui.component.DefaultButton
+import org.jetbrains.jewel.ui.component.Divider
+import org.jetbrains.jewel.ui.component.Text
+import org.jetbrains.jewel.ui.component.TextField
+import org.jetbrains.jewel.ui.component.Typography
+import slack.tooling.projectgen.CheckboxElement
+import slack.tooling.projectgen.DividerElement
+import slack.tooling.projectgen.SectionElement
+import slack.tooling.projectgen.SlackDesktopTheme
+import slack.tooling.projectgen.TextElement
+
+class CircuitGenUi {
+  @Stable
+  interface Events {
+    fun doCancelAction()
+  }
+
+  fun createPanel(
+    directory: Path,
+    packageName: String?,
+    baseTestClass: Map<String, String?>,
+    events: Events,
+    onFileGenerated: FileGenerationListener,
+    generationMode: GenerationMode,
+  ): JComponent {
+    return ComposePanel().apply {
+      setContent {
+        CircuitGenApp(
+          directory,
+          packageName,
+          baseTestClass,
+          events,
+          onFileGenerated,
+          generationMode,
+        ) { content ->
+          SlackDesktopTheme(content = content)
+        }
+      }
+    }
+  }
+
+  @Composable
+  internal fun CircuitGenApp(
+    directory: Path,
+    packageName: String?,
+    baseTestClass: Map<String, String?>,
+    events: Events,
+    onFileGenerated: FileGenerationListener,
+    generationMode: GenerationMode,
+    render: @Composable (content: @Composable () -> Unit) -> Unit,
+  ) {
+    val circuit = remember {
+      Circuit.Builder()
+        .addPresenterFactory { _, _, _ ->
+          CircuitGenPresenter(
+            directory,
+            packageName,
+            baseTestClass,
+            onFileGenerated,
+            events::doCancelAction,
+            generationMode,
+          )
+        }
+        .addUiFactory { _, _ ->
+          ui<CircuitGenScreen.State> { state, modifier -> CircuitGen(state, modifier) }
+        }
+        .build()
+    }
+    render { CircuitContent(CircuitGenScreen, circuit = circuit) }
+  }
+}
+
+@Composable
+internal fun CircuitGen(state: CircuitGenScreen.State, modifier: Modifier = Modifier) {
+
+  Box(modifier.fillMaxSize().background(JewelTheme.globalColors.paneBackground)) {
```

Comment:
Is the manual background() necessary here? I assume intellij provides a default one in the panel window

Reply:
removed!